src/channel.rs

```diff
@@ -268,176 +1423,1026 @@ impl FromXml for Channel {
+    ///         .skip_days(Vec::new())
+    ///         .items(Vec::new())
+    ///         .finalize();
+    /// ```
+    pub fn finalize(&self) -> Result<Channel, Error>
+    {
+        let mut skip_hours: Vec<String> = Vec::new();
+        for hour in self.skip_hours
+                        .clone() {
+            skip_hours.push(hour.to_string());
+        }
+
+        let ttl = match self.ttl {
+            None => None,
+            Some(val) => Some(val.to_string()),
+        };
+
+        Ok(Channel { title: self.title
+                                .clone(),
+                     link: self.link
+                               .clone(),
+                     description: self.description
+                                      .clone(),
+                     language: self.language
+                                   .clone(),
+                     copyright: self.copyright
+                                    .clone(),
+                     managing_editor: self.managing_editor
+                                          .clone(),
+                     webmaster: self.webmaster
+                                    .clone(),
+                     pub_date: self.pub_date
+                                   .clone(),
+                     last_build_date: self.last_build_date
+                                          .clone(),
+                     categories: self.categories
+                                     .clone(),
+                     generator: self.generator
+                                    .clone(),
+                     docs: self.docs
+                               .clone(),
+                     cloud: self.cloud
+                                .clone(),
+                     ttl: ttl,
+                     image: self.image
+                                .clone(),
+                     text_input: self.text_input
+                                     .clone(),
+                     skip_hours: skip_hours,
+                     skip_days: self.skip_days
+                                    .clone(),
+                     items: self.items
+                                .clone(),
+                     itunes_ext: self.itunes_ext
+                                     .clone(),
+                     dublin_core_ext: self.dublin_core_ext
+                                          .clone(),
+                     extensions: self.extensions
+                                     .clone(),
+                     namespaces: self.namespaces
+                                     .clone(), })
+    }
+}
+
+/// Enumerations of protocols for `SkipDays`.
+enum Day
+{
+    /// Monday
+    Monday,
+
+    /// Tuesday
+    Tuesday,
+
+    /// Wednesday
+    Wednesday,
+
+    /// Thursday
+    Thursday,
+
+    /// Friday
+    Friday,
+
+    /// Saturday
+    Saturday,
+
+    /// Sunday
+    Sunday,
+}
+
+impl Day
+{
+    /// Convert `&str` to `Day`.
+    pub fn value_of(s: &str) -> Result<Day, Error>
```

Comment:
This could probably implement [`FromStr`](https://doc.rust-lang.org/std/str/trait.FromStr.html) as the conversion method.